File: packages/ts/src/modules/gpu.ts
import type { HubflyClient } from '../client.js';
import type { ApiResponse, GpuInstance } from '../types.js';

export class GpuModule {
  constructor(private readonly client: HubflyClient) {}

  /**
   * List GPU instances
   */
  public async list(): Promise<ApiResponse<GpuInstance[]>> {
    return this.client.request<GpuInstance[]>('/api/v1/gpu');
  }
}

File: packages/go/gpu.go
package hubfly

import "context"

type GpuService struct {
	client *Client
}

func (s *GpuService) List(ctx context.Context) (*Response[[]GpuInstance], error) {
	return Request[[]GpuInstance](ctx, s.client, "GET", "/api/v1/gpu", nil)
}
